internal/devpkg/narinfo_cache.go

```diff
@@ -107,13 +108,30 @@ func (p *Package) fetchNarInfoStatusOnce(output string) (bool, error) {
 	type inCacheFunc func() (bool, error)
 	f, ok := narInfoStatusFnCache.Load(p.Raw)
 	if !ok {
-		key := fmt.Sprintf("%s^%s", p.Raw, output)
 		f = inCacheFunc(sync.OnceValues(func() (bool, error) { return p.fetchNarInfoStatus(output) }))
-		f, _ = narInfoStatusFnCache.LoadOrStore(key, f)
+		f, _ = narInfoStatusFnCache.LoadOrStore(p.keyForOutput(output), f)
 	}
 	return f.(inCacheFunc)()
 }
 
+func (p *Package) keyForOutput(output string) string {
+	if output == useDefaultOutput {
+		sysInfo, err := p.sysInfoIfExists()
+		// let's be super safe to always avoid empty key.
+		if err == nil && sysInfo != nil && len(sysInfo.DefaultOutputs()) > 0 {
+			output = lo.Reduce(
```

Comment:
Do the default outputs needed to be sorted to make the key stable?

Reply:
The implementation of `DefaultOutputs` is pretty stable, but it doesn't hurt.